code/data_structures/src/stack/balanced_expression/balanced_expression.cpp

```diff
@@ -0,0 +1,70 @@
+#include <iostream>
+#include <vector>
+#include <cstring>
+#include <stack>
+
+using namespace std;
+
+bool isOpen(char c)
+{
+    if(c == '(' || c == '{' || c == '[')
+        return true;
+    return false;
+}
+
+bool isBalanced(char popped, char now)
+{
+    if(popped == '(' && now ==')')
```

Comment:
```C++
return (popped == '(' && now ==')') ||
    (popped == '[' && now ==']') ||
    (popped == '{' && now =='}');
```

Reply:
As the return type is `bool` so we need to return `true` or `false` and not `1` or `0`. I have checked that the code fails in that case.

Reply:
@bahl24 Sorry what? Last I checked, the or operator returns a `bool` value. `1` and `0` evaluate to `true` and `false` respectively anyway. This shouldn't be a problem, unless the return type was that of an integer (it isn't). Could you clarify what test cases this fails in?

Reply:
Sample input- `(})`
It fails to output "NO" when we use `return (popped == '(' && now ==')') ||
    (popped == '[' && now ==']') ||
    (popped == '{' && now =='}');`
But, it shouldn't be a problem as the latest code passes any sample input given and is more easy to understand.

Reply:
Actually, it does output `NO`. Check this and modify as you desire: https://wandbox.org/permlink/ygACUgdFsVMLdG8Q

As for your second point, I personally disagree, since it is repetitive, but it is a matter of opinion so I'm not going to contradict you there. However, the code I gave _definitely_ works, as both snippets are logically equivalent, as shown in the wandbox above (I used the sample input you gave).